features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2

```diff
@@ -887,6 +887,14 @@
                     {% endfor %}
                 </AllowedSignatureAlgorithms>
             </FAPI>
+            {% if oauth.oidc.subject_type.default_subject_type is defined %}
+                <DefaultSubjectType>{{oauth.oidc.subject_type.default_subject_type}}</DefaultSubjectType>
+            {% else %}
+                <DefaultSubjectType>public</DefaultSubjectType>
+            {% endif %}
```

Comment:
Lets add this via the default.json then we can remove the conditional check.

Reply:
fixed with [f2bd155](https://github.com/wso2/carbon-identity-framework/pull/4908/commits/f2bd15540efd6caff6fd5220da54ea527d2ad59d)